kernel/arch/dreamcast/include/dc/syscalls.h

```diff
@@ -0,0 +1,395 @@
+
+    \return                 0 on success, or non-zero on
+                            failure.
+*/
+int syscall_gdrom_dma_transfer(uint32_t id, int *params);
+
+/** \brief   Checks a GDROM DMA transfer
+    \ingroup system_calls
+
+    This function checks the progress of a DMA transfer for 
+    CMD_DMAREAD_STREAM_EX (see cdrom.h).
+
+    \param  id              The request id (>=0).
+    \param  size            The pointer to receive the remaining amount of
+                            bytes to transfer.
+
+    \retval 0               On success.
+    \retval -1              On failure.
+*/
+int syscall_gdrom_dma_check(uint32_t id, int *size);
+
+/** \brief   Setup GDROM PIO callback
+    \ingroup system_calls
+
+    This function sets up PIO transfer end callback for 
+    CMD_PIOREAD_STREAM_EX (see cdrom.h).
+
+    \param  callback        The function to call upon completion of the
+                            transfer.
+    \param  param           The data to pass to the callback function.
+*/
+void syscall_gdrom_pio_callback(uintptr_t callback, void *param);
+
+/** \brief   Initiates a GDROM PIO transfer
+    \ingroup system_calls
+
+    This function initiates a PIO transfer for 
+    CMD_PIOREAD_STREAM_EX (see cdrom.h).
+
+    \param  id              The request id (>=0).
+    \param  params          The pointer to two 32-bit integers. The first 
+                            element indicates the destination address, and 
+                            the second element identifies how many bytes to 
+                            transfer.
+
+    \return                 0 on success, or non-zero on
+                            failure.
+*/
+int syscall_gdrom_pio_transfer(uint32_t id, int *params);
+
+/** \brief   Checks a GDROM PIO transfer
+    \ingroup system_calls
+
+    This function checks the progress of a PIO transfer for 
+    CMD_PIOREAD_STREAM_EX (see cdrom.h).
+
+    \param  id              The request id (>=0).
+    \param  size            The pointer to receive the remaining amount of
+                            bytes to transfer.
+
+    \retval 0               On success.
+    \retval -1              On failure.
+*/
+int syscall_gdrom_pio_check(uint32_t id, int *size);
+
+/** \brief   Clear user defined vectors.
+    \ingroup system_calls
+
+    This function clears all the user defined syscall vectors.
+
+    \return                 0
+*/
+int syscall_misc_init(void);
+
+/** \brief   Register/Clear a user defined vector
+    \ingroup system_calls
+
+    This function sets/clears the handler for one of the seven user defined
+    vectors. Setting a handler is only allowed if it not currently set.
+
+    \param  vector          The vector number (1-7).
+    \param  handler         The pointer to handler function, or NULL to
+                            clear.
+
+    \retval 0               On success.
+    \retval -1              On failure.
+*/
+int syscall_misc_setvector(uint8_t super, uintptr_t handler);
+
```

Comment:
```suggestion

/** @} */ 

```
This ends the syscall grouping.